.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDateTimePeriodParser.cs

```diff
@@ -126,15 +126,31 @@ private DateTimeResolutionResult MergeDateAndTimePeriod(string text, DateObject
             var futureDate = (DateObject)((DateTimeResolutionResult)pr1.Value).FutureValue;
             var pastDate = (DateObject)((DateTimeResolutionResult)pr1.Value).PastValue;
 
+            // handle cases with time like 25時 which resolve to the next day
+            var swiftDay = 0;
+            var timexHours = TimexUtility.ParseHoursFromTimePeriodTimex(pr2.TimexStr);
+            if (timexHours.Item1 > Constants.DayHourCount)
+            {
+                pastDate = pastDate.AddDays(1);
+                futureDate = futureDate.AddDays(1);
+            }
+            else if (timexHours.Item2 > Constants.DayHourCount)
+            {
+                swiftDay++;
+            }
+
+            var pastDate1 = pastDate.AddDays(swiftDay);
```

Comment:
Maybe rename to pastDateAlt as it's an alternative resolution?

Reply:
done